devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -139,13 +139,9 @@ public void execute(Task test) {
                     compileJavaTask.dependsOn(quarkusGenerateCode);
                     quarkusGenerateCode.setSourceRegistrar(compileJavaTask::source);
 
-                    Task compileKotlinTask = tasks.findByName("compileKotlin");
-                    // TODO: proper support of kotlin
-                    if (compileKotlinTask == null) {
-                        JavaCompile compileTestJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
-                        compileTestJavaTask.dependsOn(quarkusGenerateCodeTests);
-                        quarkusGenerateCodeTests.setSourceRegistrar(compileTestJavaTask::source);
-                    }
+                    JavaCompile compileTestJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
+                    compileTestJavaTask.dependsOn(quarkusGenerateCodeTests);
+                    quarkusGenerateCodeTests.setSourceRegistrar(compileTestJavaTask::source);
```

Comment:
maybe we should do the same in the kotlin plugin block but I don't know if can add the `kotlin.jvm` plugin as a project dependency to get the type of `compileKoltin` task.

Reply:
Yeah, not sure if adding a hard dependency on the kotlin plugin is a good idea in this case. We could revisit that after this issue is resolved